### PR TITLE
CodeQL: remove import linter hack

### PR DESCRIPTION
**Pull request by bryevdv**
CodeQL reports a great number of "Statement has no effect` warnings for our use of this hacky method to suppress linter warnings in the past:
```python
from .property.vectorization import expr; expr
```
But this hack is not necessary at all if the import is included in `__all__` and in fact, our use of the hack only served to prevent contemporary linters from telling us when we failed to to keep `__all__` up to date, which we had several cases of. 

This PR removes the import linter hack throughout, fixes up `__all__` values that were missing entries, and also removes `script/spec.py` which was discovered to be non-functional.

### CODE REVIEW

File: tests/unit/bokeh/plotting/test___init___plotting.py [hunk 1]
```diff
@@ -42,6 +43,7 @@
     'output_notebook',
     'reset_output',
     'row',
+    'Row',
```
Comment (bryevdv) [inline tests/unit/bokeh/plotting/test___init___plotting.py:46]: alternatively, we could remove `Row` and `Column` entirely, but that would have been better done at 3.0

File: tests/support/defaults.py [hunk 2]
```diff
@@ -83,6 +81,12 @@ def collect_defaults() -> dict[str, Any]:
     serializer = DefaultsSerializer()
     defaults: dict[str, Any] = {}
 
+    # in order to look up from the model catalog that Model maintains, it
+    # has to be created first. These imports ensure that all built-in Bokeh
+    # models are represented in the catalog.
+    import bokeh.models  # noqa: F401
+    from bokeh.plotting import GMap, figure  # noqa: F401
```
Comment (mattpap) [inline tests/support/defaults.py]: Wouldn't `import bokeh.plotting` suffice?

File: tests/support/defaults.py [hunk 3]
```diff
@@ -83,6 +81,12 @@ def collect_defaults() -> dict[str, Any]:
     serializer = DefaultsSerializer()
     defaults: dict[str, Any] = {}
 
+    # in order to look up from the model catalog that Model maintains, it
```
Comment (mattpap) [inline tests/support/defaults.py]:
```suggestion
    # In order to look up from the model catalog that Model maintains, it
```

File: tests/support/defaults.py (hunk 2 above)
Comment (bryevdv) [reply tests/support/defaults.py]: https://github.com/bokeh/bokeh/pull/12645/commits/89c48ae49e3fcd49777d1a6f2e6f4dfa33ef4c51

File: tests/support/defaults.py (hunk 3 above)
Comment (bryevdv) [reply tests/support/defaults.py]: https://github.com/bokeh/bokeh/pull/12645/commits/89c48ae49e3fcd49777d1a6f2e6f4dfa33ef4c51

File: tests/unit/bokeh/plotting/test___init___plotting.py (hunk 1 above)
Comment (bryevdv) [reply tests/unit/bokeh/plotting/test___init___plotting.py:46]: Let's keep the status quo for now. If anyone feels strongly about removing these from `bokeh.plotting` please make an issue.

**Pull request merged**